components/server/src/authorization/spicedb-authorizer.ts

```diff
@@ -196,3 +228,111 @@ export class SpiceDBAuthorizerImpl implements SpiceDBAuthorizer {
         }) as any as grpc.Metadata;
     }
 }
+
+type ZedTokenCacheKV = [objectRef: v1.ObjectReference | undefined, token: string | undefined];
+interface ZedTokenCache {
+    get(objectRef: v1.ObjectReference): Promise<string | undefined>;
+    set(...kvs: ZedTokenCacheKV[]): Promise<void>;
+    consistency(resourceRef: v1.ObjectReference | undefined): Promise<v1.Consistency>;
+}
+
+type ContextWithZedToken = RequestContext & { zedToken?: StoredZedToken };
+function getContext(): ContextWithZedToken {
+    return getRequestContext() as ContextWithZedToken;
+}
+
+/**
+ * This is a simple implementation of the ZedTokenCache that uses the local context to store single ZedToken per API request, which is stored in AsyncLocalStorage.
+ * To make this work we make the "assumption" that ZedTokens string (meant to be opaque) represent a timestamp which we can order. This is at least true for the MySQL datastore we are using.
+ */
+export class RequestLocalZedTokenCache implements ZedTokenCache {
+    constructor() {}
+
+    async get(objectRef: v1.ObjectReference | undefined): Promise<string | undefined> {
+        return getContext().zedToken?.token;
+    }
+
+    async set(...kvs: ZedTokenCacheKV[]) {
```

Comment:
wouldn't it be enough to just pass in a single token and only set it if it is greater than the existing one?

Reply:
We get a bunch of tokens in some cases. As mentioned above, happy to remove all of this once we decided the current implementation is sufficient.

Reply:
only because you want to update them per resource which we currently don't do/need